cirq/circuits/circuit_operation_test.py

```diff
@@ -0,0 +1,508 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest, sympy
+
+import cirq
+
+
+def test_properties():
+    a, b, c = cirq.LineQubit.range(3)
+    circuit = cirq.FrozenCircuit(
+        cirq.X(a),
+        cirq.Y(b),
+        cirq.H(c),
+        cirq.CX(a, b) ** sympy.Symbol('exp'),
+        cirq.measure(a, b, c, key='m'),
+    )
+    op = cirq.CircuitOperation(circuit)
```

Comment:
Maybe add a `to_op` method on frozen circuit so can do `op = circuit.freeze().to_op()`

Reply:
I'd prefer to keep the `CircuitOperation` creation separate for a couple reasons:

1. `FrozenCircuits` are meant to be reusable to improve serialization; `circuit.freeze().to_op()` lends to a 1:1 relation between `FrozenCircuit` and `CircuitOperation`
2. Creating `CircuitOperation` in `FrozenCircuit` adds import-ordering confusion, which is already a little weird due to defining `CircuitOperation` in a hierarchy that prevents `ops` from importing `circuits`.

Reply:
Funny, I had the same idea as @maffoo - a `FrozenCircuit.on(qubits)` could return a `CircuitOperation`. Maybe we don't need `on(qubits)` just `to_op()`...

1. not sure I follow - `op = c.to_op(); [c.to_op(),c.to_op(), op, op]` will have the same exact `FrozenCircuit` instance all four times, just with 3 different `CircuitOperation` instances.
2. local import can workaround that issue 
This thing works quite well: 
```
    def to_op(self):
         from cirq.circuits import CircuitOperation
         return CircuitOperation(self)
```
This is not urgent though, we can follow-up after this PR.

Reply:
Done.